messages_appendix.cddl

```diff
@@ -549,7 +505,159 @@ receive-video-capability = {
   ? 8: bool ; supports-scaling
 }
 
-receive-data-capability = {
-  0: format ; format
+; type key 134
+streaming-session-start-request = {
+  request
+  1: uint; streaming-session-id
+  2: [* media-stream-offer] ;  stream-offers
+  3: microseconds ; stats-interval
+}
+
+; type key 135
+streaming-session-start-response = {
+  response
+  1: &result ; result
+  2: [* media-stream-request] ; stream-requests
+  3: microseconds ; stats-interval
+}
+
+; type key 136
+streaming-session-modify-request = {
+  request
+  1: uint ; streaming-session-id
+  2: [* media-stream-request] ; stream-requests
+}
+
+; type key 137
+streaming-session-modify-response = {
+  response
+  1: &result ; result
+}
+
+; type key 138
+streaming-session-terminate-request = {
+  request
+  1 : uint ; streaming-session-id
+}
+
+; type key 139
+streaming-session-terminate-response = {
+  response
+}
+
+; type key 140
+streaming-session-terminate-event = {
+  1 : uint ; streaming-session-id
+}
+
+media-stream-offer = {
+  1 : uint ; media-stream-id
+  ? 2 : text ; friendly-name
+  ? 3 : [1* audio-encoding-offer] ; audio
+  ? 4 : [1* video-encoding-offer] ; video
+}
+
+media-stream-request = {
+  1 : uint ; media-stream-id
+  ? 2 : audio-encoding-request ; audio
+  ? 3 : video-encoding-request ; video
+}
+
+audio-encoding-offer = {
+  1 : uint ; encoding-id
+  2 : text ; codec
+  3 : uint ; time-scale
+  ? 4 : uint ; default-duration
+}
+
+video-encoding-offer = {
+  1 : uint ; encoding-id
+  2 : text ; codec
+  3 : uint ; time-scale
+  ? 4 : uint ; default-duration
+  ? 5 : video-rotation ; default-rotation
+}
+
+audio-encoding-request = {
+  1: uint ; encoding-id
+}
+
+video-encoding-request = {
+  1 : uint ; encoding-id
+  ? 2 : video-resolution ; target-resolution
+  ? 3 : float32 ; max-frames-per-second
+}
+
+; type key 13
+audio-frame = [
+  encoding-id: uint
+  start-time: uint
+  payload: bytes
+  ? optional: {
+    ? 0: uint ; duration
+    ? 1: media-time ; sync-time
+  }
+]
+
+; type key 14
+video-frame = {
+  0: uint ; encoding-id
+  1: uint ; sequence-number
+  ? 2: [* int] ; depends-on
+  3: uint ; start-time
+  ? 4: uint ; duration
+  5: bytes ; payload
+  ? 6: uint ; rotation
+  ? 7: media-time ; sync-time
+}
+
+; type key 141
+video-frame-request = {
+ 1: uint; encoding-id
+ 2: uint; sequence-number
+ ? 3: uint; highest-decoded-frame-sequence-number
+}
+
+ratio = [
+  antecedent: uint
+  consequent: uint
+]
+
+
+; type key 142
+streaming-session-sender-stats-event = {
+  1: uint; streaming-session-id
+  2: microseconds ; system-time
+  ? 3: [1* {
```

Comment:
Why is this dictionary nested in an array (versus just being a nested dictionary)? 
Same comment applies to other arrays here.

Reply:
Because there can be many: one for each encoding ID.